metalcloud/data_source_metalcloud_extension.go

```diff
@@ -0,0 +1,112 @@
+package metalcloud
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
```

Comment:
Let's move all constants to a `constants.go` file, so we can reuse them in different resources.
Also, let's avoid underscores and use camel case (see https://go.dev/doc/effective_go#mixed-caps)
To avoid exporting the constants outside our package we need the name to start with lowercase prefix (e.g. `const`)